### PR TITLE
Move the parallelized mapping outside db transaction

### PR DESCRIPTION
**Pull request by LZRS**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #[issue number]

**Description**
Moved the parallized parsing of resources fetched from DB to be done outside the db trnsaction lambda

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: (Bug fix | Feature | Documentation | Testing | Code health | Builds | Releases | Other)

**Screenshots (if applicable)**

**Checklist**

- [ ] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/CODE_OF_CONDUCT.md).
- [ ] I have read the [Contributing](https://google.github.io/android-fhir/contrib/) page.
- [ ] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [ ] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [ ] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [ ] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: engine/src/main/java/com/google/android/fhir/db/impl/DatabaseImpl.kt [hunk 1]
```diff
@@ -229,49 +229,54 @@ internal class DatabaseImpl(
...
       }
+
+    return dbResult.pmap(
+      Dispatchers.Default,
+    ) {
+      ResourceWithUUID(
+        it.uuid,
+        FhirContext.forR4Cached().newJsonParser().parseResource(it.serializedResource) as R,
+      )
     }
```
Comment (FikriMilano) [inline engine/src/main/java/com/google/android/fhir/db/impl/DatabaseImpl.kt:244]: The updated code adds more readability, which is nice. But I'm ok w the previous too.
I've no strong opinion about this

Comment (jingtang10) [reply engine/src/main/java/com/google/android/fhir/db/impl/DatabaseImpl.kt:244]: were you going to make a suggestion?

File: engine/src/main/java/com/google/android/fhir/db/impl/DatabaseImpl.kt [hunk 2]
```diff
@@ -229,49 +229,54 @@ internal class DatabaseImpl(
...
-    return db.withTransaction {
-      resourceDao.getResources(SimpleSQLiteQuery(query.query, query.args.toTypedArray())).pmap(
-        Dispatchers.Default,
-      ) {
-        ResourceWithUUID(
-          it.uuid,
-          FhirContext.forR4Cached().newJsonParser().parseResource(it.serializedResource) as R,
-        )
+    val dbResult =
+      db.withTransaction {
```
Comment (jingtang10) [inline engine/src/main/java/com/google/android/fhir/db/impl/DatabaseImpl.kt:233]: @aditya-07 pointed out that search might not need to be in transaction

File: engine/src/main/java/com/google/android/fhir/db/impl/DatabaseImpl.kt [hunk 3]
```diff
@@ -229,49 +229,54 @@ internal class DatabaseImpl(
...
   ): List<ResourceWithUUID<R>> {
-    return db.withTransaction {
-      resourceDao.getResources(SimpleSQLiteQuery(query.query, query.args.toTypedArray())).pmap(
-        Dispatchers.Default,
-      ) {
-        ResourceWithUUID(
-          it.uuid,
-          FhirContext.forR4Cached().newJsonParser().parseResource(it.serializedResource) as R,
-        )
+    val dbResult =
```
Comment (jingtang10) [inline engine/src/main/java/com/google/android/fhir/db/impl/DatabaseImpl.kt:232]: not feeling super strongly about this but:

would it make the code a bit more succint to just remove this variable - and chain the pmap function call.

same with the rest of the pr

**Review by jingtang10 — APPROVED** [review]

**Pull request merged**